spec/services/git_based_domain_import_service_spec.rb

```diff
@@ -1,66 +1,87 @@
 describe GitBasedDomainImportService do
-  describe "#import" do
-    let(:git_repo) { double("GitRepository", :git_branches => git_branches, :id => 123) }
-    let(:domain) { double("MiqAeDomain") }
+  let(:git_repo) { double("GitRepository", :git_branches => git_branches, :id => 123) }
+  let(:domain) { double("MiqAeDomain") }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:task) { FactoryGirl.create(:miq_task) }
+  let(:ref_name) { 'the_branch_name' }
+  let(:ref_type) { 'branch' }
+  let(:task_options) { {:action => "Import git repository", :userid => user.userid} }
+  let(:queue_options) do
+    {
+      :class_name  => "MiqAeDomain",
+      :method_name => "import_git_repo",
+      :role        => "git_owner",
+      :args        => [import_options]
+    }
+  end
+  let(:import_options) do
+    {
+      "git_repository_id" => git_repo.id,
+      "ref"               => ref_name,
+      "ref_type"          => ref_type,
+      "tenant_id"         => 321
+    }
+  end
```

Comment:
I think I'd rather see all of this setup either contained within the `#import` describe block, or using a shared context. They're `let`s, so they're lazy and won't get used if they aren't called, but it feels odd when they're set up outside of a describe they aren't used in.

I understand there's commonality between `#import` and `#queue_import`, but since all of it is shared, then we should probably use a shared context:

```
shared_context "some descriptive name" do
  let(:one) { 1 }
  let(:two) { "two" }
  # ... etc
end

describe "#my_fancy_method" do
  include_context "some descriptive name"

  it "does stuff" do
    # ... etc
  end
end
```

However, was your intention that `#queue_import` actually be public? It seems like we only want `#import` to be exposed to the public, and that we should make `#queue_import` private, and then the spec would already be covering `#import`.